plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -198,6 +219,11 @@ private Stream<DeltaLakeSplit> getSplits(
                         return Stream.empty();
                     }
 
+                    // no need to rewrite small file that is the only one in its partition
+                    if (isOptimize && !partitionKeysWithMoreThanOneFile.contains(getPartitionKey(originalPartitionColumns, addAction)) && maxScannedFileSizeInBytes.isPresent() && addAction.getSize() < maxScannedFileSizeInBytes.get()) {
```

Comment:
```suggestion
if (isOptimize && partitionKeysWithSingleFile.contains(getPartitionKey(originalPartitionColumns, addAction)) && maxScannedFileSizeInBytes.isPresent() && addAction.getSize() < maxScannedFileSizeInBytes.get()) {
```

The positive tense (instead of `!`) is a bit easier to follow from maintainer perspective.